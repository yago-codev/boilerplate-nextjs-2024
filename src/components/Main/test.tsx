import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render heading', () => {
    // renderizar o componente
    const { container } = render(<Main />)

    // busca o elemento e verifica a existência
    expect(screen.getByText(/react avançado/i)).toBeInTheDocument()

    // gerar snapshot
    expect(container.firstChild).toMatchSnapshot()
  })
})
