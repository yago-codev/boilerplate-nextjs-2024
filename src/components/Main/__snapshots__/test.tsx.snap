// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render heading 1`] = `
<main
  class="sc-beySPh hJUAIv"
>
  <h1>
    React Avan√ßado
  </h1>
</main>
`;
